crates/evm/traces/src/identifier/local.rs

```diff
@@ -24,21 +29,40 @@ impl<'a> LocalTraceIdentifier<'a> {
         self.known_contracts
     }
 
-    fn find_contract_from_bytecode(
-        &mut self,
-        code: &[u8],
-    ) -> Option<(&'a ArtifactId, &'a JsonAbi)> {
-        self.known_contracts
-            .iter()
-            .filter_map(|(id, (abi, known_code))| {
-                // Note: the diff score can be inaccurate for small contracts so we're using
-                // a relatively high threshold here to avoid filtering out too many
-                // contracts.
-                let score = bytecode_diff_score(known_code, code);
-                (score < 0.85).then_some((score, id, abi))
-            })
-            .min_by_key(|(score, _, _)| OrderedFloat(*score))
-            .map(|(_, id, abi)| (id, abi))
+    /// Iterates over artifacts with code length less than or equal to the given code and tries to
+    /// find a match.
+    ///
+    /// We do not consider artifacts with code length greater than the given code length as it is
+    /// considered that after compilation code can only be extended by additional parameters
+    /// (immutables) and cannot be shortened.
+    pub fn identify_code(&self, code: &[u8]) -> Option<(&'a ArtifactId, &'a JsonAbi)> {
+        let mut min_score = f64::MAX;
+        let mut min_score_id = None;
+
+        let ids_start = match self
+            .ordered_ids
+            .binary_search_by(|(_, known_code_len)| known_code_len.cmp(&code.len()))
+        {
+            // Exact match.
+            Ok(i) => i,
+            // Not found, start searching from the previous index.
+            Err(i) => i.saturating_sub(1),
+        };
+        for &(id, _) in &self.ordered_ids[ids_start..] {
+            let (abi, known_code) = self.known_contracts.get(id)?;
```

Comment:
are we iterating here over artifacts with length >= `code.len()` or I am missing something?

we should probably account for the same 10% delta here and start from a bit earlier position

Reply:
> are we iterating here over artifacts with length >= code.len()

Yes

> we should probably account for the same 10% delta here and start from a bit earlier position

Are you sure? Bytecode can't be smaller right, as per first PR

Reply:
I've created first PR with CWIA thing in mind (on-chain bytecode being bigger than artifact due to appended data), so the idea was to only check local bytecodes with _smaller_ size

as I understand, the only thing we consider now is metadata hash and with it we might have both situations (local code being shorter/longer) in case local and on-chain contract was compiled with different settings

so it probably should be something like `[code.len() * 0.9; code.len() + 32]`